### PR TITLE
Return builder after relations

### PR DESCRIPTION
**Pull request by BertvanHoekelen**
As discussed in #409. 

I had to put it before the `Mixins` pipe. Otherwise the `Builder` mixin on `Relation` would handle it and stop the pipe from executing.

The only thing that doesn't work is `(new User())->accounts()->whereFoo('bar');`. I didn't want to copy the code from `BuilderDynamicWheres` so let's discuss how to handle that.

### CODE REVIEW

**Comment by canvural** [comment]
Thank you! It's great.

But I was wrong in my comment [here](https://github.com/nunomaduro/larastan/issues/409#issuecomment-571968706). `$user->accounts()->where('foo', 'bar')` should return the relation, not the builder. Same for `$user->accounts()->whereIn('id', [1, 2, 3])` I tested this in Tinker in a live project, and indeed the relation is returned not the builder.

Can you fix that?

**Comment by BertvanHoekelen** [comment]
Done!

Do you have any suggestions on how we can handle those dynamic wheres? Or should we leave it for now and create an issue as feature request?

File: src/Methods/Pipes/BuilderAfterRelations.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * @param class-string $builderClass
+     */
+    private function getMethodReflectionForBuilder(PassableContract $passable, string $builderClass): ?EloquentBuilderMethodReflection
+    {
+        $eloquentBuilder = $passable->getBroker()->getClass($builderClass);
+
+        if (! $eloquentBuilder->hasNativeMethod($passable->getMethodName())) {
```
Comment (canvural) [inline src/Methods/Pipes/BuilderAfterRelations.php]: You can do an additional check here like `! Str::startsWith($passable->getMethodName(), 'where')` (not sure of the boolean logic. You can check) So we will allow the dynamic wheres also.

Comment (BertvanHoekelen) [reply src/Methods/Pipes/BuilderAfterRelations.php]: Yeah that won't work. You will then get the following error:
 `Method Illuminate\\Database\\Eloquent\\Model::whereFoo() invoked with 1 parameter, 0 required.`. 

If you take a look at [`BuilderDynamicWheres`](https://github.com/nunomaduro/larastan/blob/master/src/Methods/Pipes/BuilderDynamicWheres.php) you will see that there is much more needed to support those dynamic where's. 

So maybe we should create a `BuilderHelper` where we can return method reflection for that dynamic where. And call that in both `BuilderDynamicWhere` as `BuilderAfterRelations`. And while writing this I think we should add support for [local macros](https://github.com/nunomaduro/larastan/blob/master/src/Methods/Pipes/BuilderLocalMacros.php) on the relationships. Which we can also add to that `BuilderHelper`

Comment (canvural) [reply src/Methods/Pipes/BuilderAfterRelations.php]: Ok, you can give that a go :+1:

Comment (BertvanHoekelen) [reply src/Methods/Pipes/BuilderAfterRelations.php]: I think this is a nice solution. `BuilderLocalMacros` is depending on the model which [as discussed](https://github.com/nunomaduro/larastan/issues/409#issuecomment-571961110) I don't have access to.

**Comment by canvural** [comment]
Looks good! Thank you!

**Pull request merged**